pikaur/main.py

```diff
@@ -77,6 +78,20 @@ def init_output_encoding() -> None:
 init_output_encoding()
 
 
+def init_proxy() -> None:
+    proxy = PikaurConfig().network.get_str('Socks5Proxy')
+    if proxy:  # pragma: no cover
+        port = 1080
+        idx = proxy.find(':')
+        if idx >= 0:
+            port = int(proxy[idx + 1:])
+            proxy = proxy[:idx]
+
+        import socks  # type: ignore # pylint: disable=import-error
```

Comment:
does importing socks module executing some code by itself so you're isolating its import?

Reply:
ah, i see, because it's optional dependency:

> SocksiPy

in that case

1) its import should be try-catched and print some error what that dep should be installed in order to use proxy
2) PKGBUILD (in root of the repo) should be also modified to include new optional dependency

Reply:
for example of optional dependency error see below in main.py how it's handling systemd version and other deps messages